QModManager/API/SMLHelper/Handlers/CraftTreeHandler.cs

```diff
@@ -0,0 +1,339 @@
+        /// <param name="displayName">The display name of the tab, which will show up when you hover your mouse on the tab.</param>
+        /// <param name="sprite">The sprite of the tab.</param>
+        /// <param name="stepsToTab">
+        /// <para>The steps to the target tab.</para>
+        /// <para>These must match the id value of the CraftNode in the crafting tree you're targeting.</para>
+        /// <para>Do not include "root" in this path.</para>
+        /// </param>        
+        void ICraftTreeHandler.AddTabNode(CraftTree.Type craftTree, string name, string displayName, UnityEngine.Sprite sprite, params string[] stepsToTab)
+        {
+            ValidateStandardCraftTree(craftTree);            
+            string modName = ReflectionHelper.CallingAssemblyByStackTrace().GetName().Name;
+
+            CraftTreePatcher.TabNodes.Add(new TabNode(stepsToTab, craftTree, new Atlas.Sprite(sprite), modName, name, displayName));
+        }
+
+        /// <summary>
+        /// <para>Removes a node at the specified node location. Can be used to remove either tabs or craft nodes.</para>
+        /// <para>If a tab node is selected, all child nodes to it will also be removed.</para>
+        /// </summary>
+        /// <param name="craftTree">The target craft tree to edit.</param>
+        /// <param name="stepsToNode">
+        /// <para>The steps to the target node.</para>
+        /// <para>These must match the id value of the CraftNode in the crafting tree you're targeting.</para>
+        /// <para>This means matching the id of the crafted item or the id of the tab name.</para>
+        /// <para>Do not include "root" in this path.</para>
+        /// </param>
+        
+        void ICraftTreeHandler.RemoveNode(CraftTree.Type craftTree, params string[] stepsToNode)
+        {
+            ValidateStandardCraftTree(craftTree);
+            CraftTreePatcher.NodesToRemove.Add(new Node(stepsToNode, craftTree));
+        }
+
+        #endregion
+
+        private static void ValidateStandardCraftTree(CraftTree.Type craftTree)
+        {
+            switch (craftTree)
+            {
+                case CraftTree.Type.Fabricator:
+                case CraftTree.Type.Constructor:
+                case CraftTree.Type.Workbench:
+                case CraftTree.Type.SeamothUpgrades:
+                case CraftTree.Type.MapRoom:
+                case CraftTree.Type.Centrifuge:
+                case CraftTree.Type.CyclopsFabricator:
+                case CraftTree.Type.Rocket:
+                    break; // Okay
+                case CraftTree.Type.Unused1:                    
+                case CraftTree.Type.Unused2:                    
+                case CraftTree.Type.None:
+                default:
+                    throw new ArgumentException($"{nameof(craftTree)} value of '{craftTree}' does not correspond to a standard crafting tree.{Environment.NewLine}" +
```

Comment:
We need to talk about `Enivonment.NewLine`

Reply:
This is for logging, not for printing on screen.  
It makes sense to use Environment.NewLine here since the text will be handled by the OS, not by Unity.

Reply:
That's why I said we need to talk. The only reason Windows still uses \r\n is for backwards compatibility. Windows doesn't care if we use \n or \r\n, it's the same thing.

Reply:
Sure, Windows can accept both, but we should try to avoid mixing line endings in the same file if we can help it.  
No reason to be _that_ sloppy.